turbot/resource_turbot_mod.go

```diff
@@ -270,6 +270,32 @@ func getLatestCompatibleVersion(org, modName, version string, meta interface{})
 		return "", err
 	}
 
+	availableModVersions := buildModVersionListByStatus(modVersions, "available")
```

Comment:
I think we can do this more simply

1) getPotentialVersions
this is basically the current code (i.e. fetch all versions, then loop through checking version/status) except it builds a list of all versions matching the version with either recommended/available (rather than just picking one)
2) getVersionWithStatus
take the list of potential versions and return the highest version with the desired status

So code is:

```
potentialVersions = getPotentialVersions()
version = getVersionWithStatus(potentialVersions, "RECOMMENDED")
if version == nil {
  version = getVersionWithStatus(potentialVersions, "AVAILABLE")
}
if (version == nil {
  ERROR
}